apps/examples/webserver_test/webserver_test_main.c

```diff
@@ -0,0 +1,1293 @@
+
+	PRNT("TLS Init Success");
+
+	if ((result = mbedtls_ssl_setup(&(client->tls_ssl),
+									&(client->tls_conf))) != 0) {
+		PRNT("Error: mbedtls_ssl_setup returned -%4x", -result);
+		goto HANDSHAKE_FAIL;
+	}
+
+	/*
+	 * Server name intication is an extension to the TLS networking protocol
+	 * If server presents multiple certificates on the same IP address,
+	 * client could make multiple secure session depends on hostname.
+	 *
+	 * Note : Hostname in TLS is a subject's common name(CN) of certificates.
+	 */
+#if WEBCLIENT_CONF_CHECK_TLS_HOSTNAME
+	if ((result = mbedtls_ssl_set_hostname(&(client->tls_ssl), hostname)) != 0) {
+		PRNT("Error: mbedtls_hostname returned -%4x", -result);
+		goto HANDSHAKE_FAIL;
+	}
+#endif
+
+	mbedtls_ssl_set_bio(&(client->tls_ssl), &(client->tls_client_fd),
+						mbedtls_net_send, mbedtls_net_recv, NULL);
+
+	/* Handshake */
+	while ((result = mbedtls_ssl_handshake(&(client->tls_ssl))) != 0) {
+		if (result != MBEDTLS_ERR_SSL_WANT_READ &&
+			result != MBEDTLS_ERR_SSL_WANT_WRITE) {
+			PRNT("Error: TLS Handshake fail returned -%4x", -result);
+			goto HANDSHAKE_FAIL;
+		}
+	}
+
+	PRNT("TLS Handshake Success");
+	FUNC_EX;
+	return 0;
+HANDSHAKE_FAIL:
+	FUNC_EX;
+	return result;
+}
+#endif
+
+static int ws_test_wget_msg_construct(char *buf, struct http_client_request_t *param, struct ws_test_wget_s *ws)
+{
+	FUNC_EN;
+	int post_len;
+	char *dest = buf;
+	char *temp;
+	char post_size[8];
+	struct http_keyvalue_t *cur = NULL;
+
+	/* Send method */
+	if (param->method == WGET_MODE_GET) {
+		dest = ws_test_wget_strcpy(dest, g_httpget, param);
+	} else if (param->method == WGET_MODE_POST) {
+		dest = ws_test_wget_strcpy(dest, g_httppost, param);
+	} else if (param->method == WGET_MODE_PUT) {
+		dest = ws_test_wget_strcpy(dest, g_httpput, param);
+	} else if (param->method == WGET_MODE_DELETE) {
+		dest = ws_test_wget_strcpy(dest, g_httpdelete, param);
+	}
+
+	if (dest == NULL) {
+		return WGET_MSG_CONSTRUCT_ERR;
+	}
+
+	dest = ws_test_wget_strcpy(dest, ws->filename, param);
+	if (dest == NULL) {
+		return WGET_MSG_CONSTRUCT_ERR;
+	}
+
+	*dest++ = ISO_space;
+	dest = ws_test_wget_strcpy(dest, g_http11, param);
+	if (dest == NULL) {
+		return WGET_MSG_CONSTRUCT_ERR;
+	}
+	dest = ws_test_wget_strcpy(dest, g_httpcrnl, param);
+	if (dest == NULL) {
+		return WGET_MSG_CONSTRUCT_ERR;
+	}
+	dest = ws_test_wget_strcpy(dest, g_httphost, param);
+	if (dest == NULL) {
+		return WGET_MSG_CONSTRUCT_ERR;
+	}
+	dest = ws_test_wget_strcpy(dest, ws->hostname, param);
+	if (dest == NULL) {
+		return WGET_MSG_CONSTRUCT_ERR;
+	}
+	dest = ws_test_wget_strcpy(dest, g_httpcrnl, param);
+	if (dest == NULL) {
+		return WGET_MSG_CONSTRUCT_ERR;
+	}
+
+	/* header of entity */
+
+	if (param->method == WGET_MODE_POST || param->method == WGET_MODE_PUT) {
+		/* Look for Content-Type in the headers */
+		cur = param->headers->head->next;
+		while (cur != param->headers->tail) {
+			if (!strncmp(cur->key, "Content-Type", strlen("Content-Type"))) {
```

Comment:
Can we please make it as "(!strncmp(cur->key, "Content-Type", strlen("Content-Type")+1))" ?

Reply:
Changes done as suggested